src/main/java/com/iridium/iridiumskyblock/bankitems/CrystalsBankItem.java

```diff
@@ -34,13 +38,45 @@ public BankResponse withdraw(Player player, Number amount, TeamBank teamBank, Ir
     }
 
     @Override
-    public BankResponse deposit(Player player, Number amount, TeamBank teamBank, IridiumTeams<?, ?> iridiumTeams) {
-        int crystals = removeCrystals(player.getInventory());
-        if (crystals > 0) {
-            teamBank.setNumber(teamBank.getNumber() + crystals);
-            return new BankResponse(crystals, true);
+    public BankResponse deposit(Player player, Number number, TeamBank teamBank, IridiumTeams<?, ?> iridiumTeams) {
+        Optional<Island> islandOptional = IridiumSkyblock.getInstance().getUserManager().getUser(player).getIsland();
+        if (!islandOptional.isPresent()) {
+            return new BankResponse(0, false);
         }
-        return new BankResponse(crystals, false);
+
+        int remainingItemAmount = number.intValue();
+        int depositAmount = 0;
+
+        ItemStack[] contents = player.getInventory().getContents();
+        for (int i = 0; i < contents.length && remainingItemAmount > 0; i++) {
+            ItemStack itemStack = contents[i];
+
+            int crystalsPerItem = IridiumSkyblock.getInstance().getIslandManager().getIslandCrystals(itemStack);
+            if (crystalsPerItem == 0) continue;
+
+            int itemStackAmount = itemStack.getAmount();
+
+            if (itemStackAmount <= remainingItemAmount) {
+                player.getInventory().setItem(i, null);
+
+                depositAmount += itemStackAmount * crystalsPerItem;
+                remainingItemAmount -= itemStackAmount;
+
+            } else {
+                itemStack.setAmount(itemStack.getAmount() - remainingItemAmount);
+                player.getInventory().setItem(i, itemStack);
+
+                depositAmount += remainingItemAmount * crystalsPerItem;
+                remainingItemAmount = 0;
+            }
```

Comment:
Im not convinced this works as intended, ill take another look later

Reply:
Surprisingly it does, I wouldn't commit code if I didn't check its efficacy first